### PR TITLE
Add sailsjs

### LINKED ISSUE #1939: Add sailsjs
Hi @mikermcneil @rachaelshaw @sgress454 @Zolmeister,

Would you mind to add (or consent that we add) [sailsjs](https://github.com/balderdashy/sails) in `here` ?

This project compare performance based on 3 routes :

+ `GET` `/`, **SHOULD** return a successful status code and an empty string
+ `GET` `/user/:id`, **SHOULD** return a successful status code and the id
+ `POST` `/user`, **SHOULD** return a successful status code and an empty string

Regards,

### PR DESCRIPTION
**Pull request by mdartic**
Related to #1939
implementation available in
https://github.com/mdartic/web-frameworks-sailsjs

------------
Closes #1939

### CODE REVIEW

File: javascript/sails/config/routes.js [hunk 1]
```diff
@@ -0,0 +1,5 @@
+module.exports.routes = {
+  "GET /": { action: "home" },
+  "GET /user/:id?": { action: "user/get" },
```
Comment (waghanza) [inline javascript/sails/config/routes.js]: is this (having `?`) means that `/user` will be allowed in `GET` ?

File: javascript/sails/config/env/development.js [hunk 2]
```diff
@@ -0,0 +1,3 @@
+module.exports = {
```
Comment (waghanza) [inline javascript/sails/config/env/development.js]: there is no **production.js** ?

File: javascript/sails/config/routes.js (hunk 1 above)
Comment (mdartic) [reply javascript/sails/config/routes.js]: Yes, you're right.
I've removed this.

File: javascript/sails/config/env/development.js (hunk 2 above)
Comment (mdartic) [reply javascript/sails/config/env/development.js]: You're right too, I rename this file in production.js

Comment (mdartic) [reply javascript/sails/config/env/development.js]: I don't know how you start the process app, but you'll have to set `NODE_ENV` to `production` to make sails consider the `production.js` file.

Comment (waghanza) [reply javascript/sails/config/env/development.js]: yep, done in `Dockerfile`

**Comment by waghanza** [comment]
@mdartic seems we have an issue => https://travis-ci.com/github/the-benchmarker/web-frameworks/jobs/300940936#L357

**Comment by mdartic** [comment]
I think it's related to production mode, with sessions in sails.
I can't see any trace in the travis log, related to the start of sails.
I've pushed some commits to disable sessions when running.

**Comment by waghanza** [comment]
when starting the application with  `node app.js`, I have
```javascript
ebug: Warning: since `sails.config.session.cookie.secure` is not set to `true`, the session
debug: cookie will be sent over non-TLS connections (i.e. with insecure http:// requests).
debug: To enable secure cookies, set `sails.config.session.cookie.secure` to `true`.
debug: 
debug: If your app is behind a proxy or load balancer (e.g. on a PaaS like Heroku), you
debug: may also need to set `sails.config.http.trustProxy` to `true`.
debug: 
debug: For more help:
debug:  • https://sailsjs.com/config/session#?the-secure-flag
debug:  • https://sailsjs.com/config/session#?do-i-need-an-ssl-certificate
debug:  • https://sailsjs.com/config/sails-config-http#?properties
debug:  • https://sailsjs.com/support
debug: 
error: Failed to lift app: Error: Session secret should be manually specified in production!
To set up a session secret, add or update it in `config/session.js`:
module.exports.session = { secret: 'extremely-secure-keyboard-cat' }

(Or if you don't need sessions enabled, disable the hook.)

For more help:
 • https://sailsjs.com/config/session
 • https://sailsjs.com/support
    at Hook.configure (/usr/src/app/node_modules/sails/lib/hooks/session/index.js:150:17)
    at Hook.wrapper [as configure] (/usr/src/app/node_modules/@sailshq/lodash/lib/index.js:3282:19)
    at /usr/src/app/node_modules/sails/lib/app/private/loadHooks.js:331:20
    at /usr/src/app/node_modules/async/dist/async.js:3083:16
    at eachOfArrayLike (/usr/src/app/node_modules/async/dist/async.js:1003:9)
    at eachOf (/usr/src/app/node_modules/async/dist/async.js:1051:5)
    at Object.eachLimit (/usr/src/app/node_modules/async/dist/async.js:3145:5)
    at configure (/usr/src/app/node_modules/sails/lib/app/private/loadHooks.js:328:17)
    at /usr/src/app/node_modules/async/dist/async.js:3853:24
    at replenish (/usr/src/app/node_modules/async/dist/async.js:946:17)
    at iterateeCallback (/usr/src/app/node_modules/async/dist/async.js:931:17)
    at /usr/src/app/node_modules/async/dist/async.js:906:16
    at /usr/src/app/node_modules/async/dist/async.js:3858:13
    at /usr/src/app/node_modules/async/dist/async.js:421:16
    at iteratorCallback (/usr/src/app/node_modules/async/dist/async.js:998:13)
    at /usr/src/app/node_modules/async/dist/async.js:906:16
```

I then :
+ [x] added a `config/session.js` file
+ [x] modified the `Dockerfile` and `config.yaml` so as to take the **config** folder in the container

**Comment by mdartic** [comment]
In fact, the `.sailsrc` file at the root of `sails` folder contains 
```
{
    "hooks": {
        "session": false
    }
}
```
and is used by `app.js`.
So, in theory, it has to be used and disable sessions ?

**Comment by waghanza** [comment]
But this file is not "uploaded" in the container

**Comment by mdartic** [comment]
Right, so do we add a new line in the `config.yaml` under the `files` section telling to copy the `.sailsrc` file ?

**Comment by waghanza** [comment]
Also, I had
~~~javascript
warn: Ignored attempt to bind route (/user/) to unknown action :: { action: 'user/post' }
 warn: Ignored attempt to bind route (/user/:id) to unknown action :: { action: 'user/get' }
 warn: Ignored attempt to bind route (/) to unknown action :: { action: 'home' }
~~~
while running the `app`

To fix it I just added `api` to `config.yaml` (in order to upload files).

There is a lack of verbosity / documentation

I'm working on new tools for that

BTW, thanks for your patience. I will merge after I have updated results (maybe tomorrow)

**Comment by waghanza** [comment]
@mdartic results has been updated

**Pull request merged**